lib/new_relic/agent/instrumentation/tilt/instrumentation.rb

```diff
@@ -6,6 +6,35 @@ module NewRelic
   module Agent
     module Instrumentation
       module Tilt
+
+        def metric_name(klass, file)
+          "View/#{klass}/#{file}/Rendering"
+        end
+
+        # Sinatra uses #caller_locations for the file name in Tilt (unlike Rails/Rack)
+        # So here we are only grabbing the file name and name of directory it is in
+        def create_filename_for_metric(file)
+          return file unless defined?(::Sinatra) && defined?(::Sinatra::Base)
+          file.split('/')[-2..-1].join('/')
+        rescue NoMethodError
+          file
+        end
+
+        def render_with_tracing(*args, &block)
+          begin
+            finishable = Tracer.start_segment(
+              name: metric_name(self.class, create_filename_for_metric(self.file))
```

Comment:
I wonder if we actually need `self` here to make it work?

Reply:
I'm not sure if it's necessary. Personally, I like having self.whatever for things we're accessing like that, it seems a little more immediately obvious to me what it's using. If you feel differently we could talk about it though!

Reply:
Nice! I don't have strong preferences toward using self or eliminating it. I was taught to use it for more clarity, but I've had people who prefer eliminating it when it isn't needed. Happy to keep it!